### PR TITLE
test: Create test for CheckPlatformReqsCommand

### PR DESCRIPTION
**Pull request by theoboldalex**
Created a test for the CheckPlatformReqsCommand which increases coverage from 8% to 76%.

### CODE REVIEW

File: tests/Composer/Test/Command/CheckPlatformReqsCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        $appTester->run(array_merge(['command' => 'check-platform-reqs'], $flag));
+
+        $appTester->assertCommandIsSuccessful();
+    }
+
+    public function flagGenerator(): \Generator
+    {
+        yield 'Disables checking of require-dev packages requirements.' => [
+            ['--no-dev' => true]
+        ];
```
Comment (Seldaek) [inline tests/Composer/Test/Command/CheckPlatformReqsCommandTest.php:35]: This is good to have but would be great to verify the behavior actually works by creating some packages and checking the output matches expected output.

See TestCase::initTempComposer, createInstalledJson  and createComposerLock for this. https://github.com/composer/composer/blob/70f2dd6eddd5916f70b3d73dd3170a3d2d3a7681/tests/Composer/Test/Command/BumpCommandTest.php#L28-L47 also has a real example of using these.

File: tests/Composer/Test/Command/CheckPlatformReqsCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        yield 'Whether to disable plugins.' => [
+            ['--no-plugins' => true]
+        ];
+
+        yield 'Checks requirements only from the lock file, not from installed packages.' => [
+            ['--lock' => true]
+        ];
```
Comment (Seldaek) [inline tests/Composer/Test/Command/CheckPlatformReqsCommandTest.php:43]: As above, would be nice to check this actually does something and not just runs without failing.

**Review by Seldaek — CHANGES_REQUESTED** [review]

**Comment by Seldaek** [comment]
Thanks for getting started here. This definitely is better than nothing as it runs the code and would catch any critical error, but it'd be cool if you find time to go further and actually check the output. See this example too for inspiration on checking output https://github.com/composer/composer/blob/70f2dd6eddd5916f70b3d73dd3170a3d2d3a7681/tests/Composer/Test/Command/UpdateCommandTest.php#L31

**Comment by theoboldalex** [comment]
Thanks for the feedback. I'll get to work on the requested changes.

**Comment by jeffwidman** [comment]
Gentle nudge @theoboldalex be great to see this move forward

**Comment by theoboldalex** [comment]
Apologies for the delay on getting back to this. I have actioned the points raised by @Seldaek so this is ready for another review. Thanks

**Review by Seldaek — APPROVED** [review]
Looks good now, just a couple style nitpicks

**Comment by Seldaek** [comment]
Thanks!

**Comment by theoboldalex** [comment]
I have actioned the code style fixes as requested.

**Pull request merged**